tests/test_catalog.py

```diff
@@ -1218,6 +1218,32 @@ def test_get_all_collections(self) -> None:
         self.assertGreater(len(all_collections), 0)
         self.assertTrue(all(isinstance(c, pystac.Collection) for c in all_collections))
 
+    def test_get_single_links_media_type(self) -> None:
+        catalog = TestCases.test_case_1()
+
+        catalog.links.append(
+            pystac.Link(rel="search", target="./search.html", media_type="text/html")
+        )
+        catalog.links.append(
+            pystac.Link(
+                rel="search", target="./search.json", media_type="application/geo+json"
+            )
+        )
+
+        html_link = catalog.get_single_link("search")
+        assert html_link is not None
+        self.assertEqual(html_link.href, "./search.html")
+        json_link = catalog.get_single_link("search", media_type="application/geo+json")
+        assert json_link is not None
+        self.assertEqual(json_link.href, "./search.json")
+
+    def test_get_links_media_type(self) -> None:
+        catalog = TestCases.test_case_9()
+        self.assertEqual(len(catalog.get_links("search")), 2)
```

Comment:
```suggestion
        catalog = TestCases.test_case_1()

        catalog.links.append(
            pystac.Link(rel="search", target="./search.html", media_type="text/html")
        )
        catalog.links.append(
            pystac.Link(
                rel="search", target="./search.json", media_type="application/geo+json"
            )
        )
```

This should fix the CI error. We probably don't need to assert then length of the links if we are adding them manually here.